x-pack/dockerlogbeat/pipelinemanager/clientLogReader.go

```diff
@@ -24,16 +26,22 @@ import (
 // There's a many-to-one relationship between clients and pipelines.
 // Each container with the same config will get its own client to the same pipeline.
 type ClientLogger struct {
-	logFile       *pipereader.PipeReader
-	client        beat.Client
-	pipelineHash  uint64
-	closer        chan struct{}
-	containerMeta logger.Info
-	logger        *logp.Logger
+	// logFile is the FIFO reader that reads from the docker container stdio
+	logFile *pipereader.PipeReader
+	// client is the libbeat client object that sends logs upstream
+	client beat.Client
+	// pipelineHash is a hash of the libbeat publisher pipeline config
+	pipelineHash uint64
+	// ContainerMeta is the metadata object for the container we get from docker
+	ContainerMeta logger.Info
+	// logger is an error message logger
+	logger *logp.Logger
+	// localLog manages the local JSON logs for containers
+	localLog logger.Logger
 }
```

Comment:
this type is kind of the core of the change. Let's group fields by functionality and add update the godoc with some more details. e.g.

```

// ClientLogger collects logs for a docker container logging to stdout and stderr, using the FIFO provided by the docker daemon.
// Each log line is written to a local log file for retrieval via "docker logs", and forwarded to the beats publisher pipeline.
// The local log storage is based on the docker json-file logger and supports the same settings. If "max-size" is not configured, we will rotate the log file every 10MB.
type ClientLogger struct {
  pipelineHash uint64  // pipeline configuration hash
  logger *logp.Logger  // for internal logging (we use the name log quite often here ;) )

  // ContainerMeta is the metadata object for the container we get from docker.
  // The metadata are added to each log event.
  ContainerMeta logger.Info

  // logFile is the FIFO reader that reads from the docker container stdio
  logFile *pipereader.PipeReader
  // client is used to publish logs to Elasticsearch
  client beat.Client
  // localLog manages the local JSON logs for the associated container
  localLog logger.Logger
}
```